### DISCUSSION TITLE
Sumatra only opens up 20 pdf files?

### DISCUSSION THREAD

**Discussion by Morgano16** [sumatrapdfreader/sumatrapdf#4278] [Q&A] [answered]
Whenever I look at a new pdf , the oldest one in my pdf collection gets discarded in sumatra. Is this a display issue or paid feature? if so im willing to pay but is there a feature in the settings or something where I can access more than 20 pddfs in Sumatra? If anybody has any advice like how to fix this, a better different pdf reader app, etc I'd appreicate it, thank you

**Comment by kjk** [comment, marked as answer]
I assume you're referring to home page which indeed has a limit of 20 thumbnails, but it's only because I didn't implement scrolling.

SumatraPDF remembers all opened files. If you invoke Command Palette `Ctrl + K`, file history is listed at the top.

**Reply by Morgano16** [reply]
Thank you so much , i didn't know about the command

**Reply by joejoe1429** [reply]
Hi, kjk sir, would you please tell me how to implement scrolling in main window so as to break the limit of showing only 20 thumbnails. Many thanks.

**Reply by GitHubRulesOK** [reply]
@joejoe1429 there is no scrolling in "main" window it is a "responsive" layout based on windows size so number of thumbnails  varies from none to maximum 20
In different ratios so has max 5 horizontal so with 4 rows that's 20 
agreed half a screen is only 12 but the internal listing is good enough
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/86eddbc7-8b20-4e40-bc6a-f6aca60e9428)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/b96e0356-bfe1-471d-976c-274fbf7fcd53)![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/adefb254-65bd-4932-a552-59cce2f6d324)

**Comment by GitHubRulesOK** [comment]
SumatraPDF is fully free to use there are no extra features unless you add your own or buy a pirates copy

There is no real limit to use, 
The "file" menu drop down shows **last 10**
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/3d70b9f5-ff3c-4e00-a857-8fafc00b9787)

"Home" screen only shows the **most frequent 20**
Favorites and Internal list (Control +K) shows no limit except it will slow you down to have too many in history
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/8fd96606-e99e-4e0a-ba42-11c1c4e5acd3)

**Reply by Morgano16** [reply]
Thank you for letting me know
